### PR TITLE
Use whoami instead of id -u to check current user ID

### PR DESCRIPTION
**Pull request by codificat**
Replace the check of the current UID based on `id -u` with `whoami` to make the purpose of the check a bit more clear / intuitive: *do I know who am I?*

### CODE REVIEW

File: starter-arbitrary-uid/bin/uid_entrypoint [hunk 1]
```diff
@@ -1,8 +1,6 @@
 #!/bin/sh
-if [ -w /etc/passwd ]; then 
-  USER_ID=$(id -u)
-  id ${USER_ID} &> /dev/null
-  if [ $? -ne 0 ]; then
+if [ -w /etc/passwd ]; then
+  if ! whoami &> /dev/null; then
```
Comment (tchughesiv) [inline starter-arbitrary-uid/bin/uid_entrypoint]: this _should_ work but `USER_ID=$(id -u)` line needs to stay i suspect.  the variable is required for the sed.

File: starter-systemd/bin/systemd_entrypoint [hunk 2]
```diff
@@ -1,8 +1,6 @@
 #!/bin/sh
 if [ -w /etc/passwd ]; then 
-  USER_ID=$(id -u)
-  id ${USER_ID} &> /dev/null
-  if [ $? -ne 0 ]; then
+  if ! whoami &> /dev/null; then
```
Comment (tchughesiv) [inline starter-systemd/bin/systemd_entrypoint]: same

**Review by tchughesiv — CHANGES_REQUESTED** [review]

**Comment by tchughesiv** [comment]
also, can you test this w/ things like rhel-atomic image, etc? i think i stuck w/ `id` originally because `whoami` isn't always installed/available.

**Comment by codificat** [comment]
Fixed the reference to `$USER_ID` (thanks) and now actually tested on AH 7.3, it works:

```
$ sudo docker run --rm -u 123456 acme/starter-arbitrary-uid id
uid=123456(default) gid=0(root) groups=0(root)
$ sudo docker run --rm -u 123456 acme/starter-arbitrary-uid:centos7 id
uid=123456(default) gid=0(root) groups=0(root)
$ sudo docker run --rm -u 12 acme/starter-arbitrary-uid id
uid=12(games) gid=100(users) groups=100(users)
$ sudo docker run --rm -u 12 acme/starter-arbitrary-uid:centos7 id
uid=12(games) gid=100(users) groups=100(users)
```

`whoami` is part of `coreutils`, same package as `id`, so I'd expect both to be present.

**Review by tchughesiv — APPROVED** [review]

**Pull request merged**